<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">



<title>cILR: Competitive Isometric Log Ratio for Taxonomic Enrichment Analysis</title>

<script src="poster_files/header-attrs-2.8/header-attrs.js"></script>





<!--
Font-awesome icons ie github or twitter
-->
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/brands.css" integrity="sha384-n9+6/aSqa9lBidZMRCQHTHKJscPq6NW4pCQBiMmHdUCvPN8ZOg2zJJTkC7WIezWv" crossorigin="anonymous">

<!--
Google fonts api stuff
-->
<link href='https://fonts.googleapis.com/css?family=Roboto' rel='stylesheet'>
<link href='https://fonts.googleapis.com/css?family=Karla' rel='stylesheet'>

<!--
Here are the required style attributes for css to make this poster work :)
-->
<style>
@page {
size: 48in 36in;
margin: 0;
padding: 0;
background-color: #ffffff;
}
body {
margin: 0px;
padding: 0px;
width: 48in;
height: 36in;
text-align: justify;
font-size: 30pt;
background-color: #FFFFFF;
color: #000000;
}
/* RMarkdown Class Styles */
/* center align leaflet map,
from https://stackoverflow.com/questions/52112119/center-leaflet-in-a-rmarkdown-document */
.html-widget {
margin: auto;
position: sticky;
margin-top: 2cm;
margin-bottom: 2cm;
}
.leaflet.html-widget.html-widget-static-bound.leaflet-container.leaflet-touch.leaflet-fade-anim.leaflet-grab.leaflet-touch-drag.leaflet-touch-zoom {
position: sticky;
width: 100%;
}
pre.sourceCode.r, pre.sourceCode.python {
background-color: #ffffff20;
border: solid ##00693e 1mm;
border-radius: 0.5em;
padding: 4mm;
margin: auto;
margin-top: 1em;
margin-bottom: 1em;
/* align-items: center; */
}
div.sourceCode {
background-color: transparent;
width: 85%;
margin-left: auto;
margin-right: auto;
}
code.sourceCode.r, code.sourceCode.python {
background-color: transparent;
font-size: 20pt;
border-radius: 2mm;
}
code {
font-size: 25pt;
font-family: monospace;
background-color: ##00693e24;
color: ##00693e;
padding: 1.2mm;
line-height: 1;
border-radius: 2mm;
}
caption {
margin-bottom: 10px;
font-size: 20pt;
font-style: italic;
}

tbody tr:nth-child(odd) {
    background-color: #00693e20;
}
.table>thead>tr>th, .table>tbody>tr>th, .table>tfoot>tr>th, .table>thead>tr>td, .table>tbody>tr>td, .table>tfoot>tr>td{
  border-spacing: 0;
  font-size: 40%;
  border-style: none;
  padding-top: 15px;
  padding-bottom: 15px;
  padding-right: 1em;
  padding-left: 1em;
  line-height: 1em;
}
table {
  margin: auto;
}
th {
  padding-left: 5mm;
  padding-right: 5mm;
}
.caption {
font-size: 20pt;
font-style: italic;
padding-top: 0;
}
.references {
font-size: 20px;
line-height: 90%;
}
/* Create three unequal columns that floats next to each other */
.column {
float: left;
padding: 0px;
}
.outer {
width: 48in;
height: 36in;
-webkit-column-count: 2; /* Chrome, Safari, Opera */
-moz-column-count: 2; /* Firefox */
column-count: 2;
-webkit-column-fill: auto;
-moz-column-fill: auto;
column-fill: auto;
-webkit-column-rule-width: 0.4;
-moz-column-rule-width: 0.4;
column-rule-width: 0.4;
-webkit-column-rule-style: solid;
-moz-column-rule-style: solid;
column-rule-style: solid;
-webkit-column-rule-color: black;
-moz-column-rule-color: black;
column-rule-color: black;
-webkit-column-gap: calc(0.4 * 100%);
background-color: #ffffff;
font-family: Karla;
color: #000000;
}
span.citation {
  color: ##00693e;
  font-weight: bold;
}
a {
text-decoration: none;
color: ##00693e;
}
#title {
font-size: 75pt;
text-align: left;
margin: 0;
line-height: 90%;
border-bottom: 0;
}
#author {
color: #00693e;
margin: 0;
text-align: left;
line-height: 65%;
font-size: 1.17em;
}
#affiliation {
padding-top: 1em;
color: #00000060;
font-style: italic;
font-size: 25px;
margin: 0;
}
sup {
color: #cc0000;
}
.affiliation sup {
font-size: 20px;
}
.author sup {
font-size: 30px;
}
.author_extra {
color: ##00693e;
margin: 0;
line-height: 85%;
font-size: 35px;
}
.outer h1 {
text-align: center;
margin-top: 0.5in;
margin-bottom: 0.5in;
}
.outer h2 {
text-align: center;
}
.outer p {
color: #000000;
}
.outer ol {
text-align: left;
}
.main {
width: calc(48in * 0.4);
height: 36in;
position: absolute;
margin-left: calc(48in * ((1 - 0.4) / 2));
background-color: #00693e;
color: #FFFFFF90;
font-family: Roboto;
background-image: linear-gradient(#00693e 50%, ##00693e);
}
.main strong {
color: #FFFFFF;
}
img.main-img-left {
width: 20%;
left: 0.5in;
bottom: 0.2in;
position: absolute;
}
img.main-img-right {
width: 18%;
right: 0.5in;
bottom: 0.2in;
position: absolute;
}
.main p {
padding-top: 20%;
font-size: 70pt;
text-align: left;
}
.fab {
color: #00000060;
font-size: 25px;
}
.twitter, i {
color: #00000060;
font-size: 35px;
text-decoration: none;
}
a.email {
text-decoration: none;
color: #00000060;
font-size: 35px;
}
.envelope {
color: #00000060;
font-size: 5px;
text-decoration: none;
}
.globe {
color: #00000060;
font-size: 5px;
text-decoration: none;
}
.github > .fab {
color: #00000060;
font-size: 40px;
text-decoration: none;
}
.globetext {
color: #00000060;
font-size: 35px;
text-decoration: none;
}
.poster_wrap {
width: 48in;
height: 36in;
padding: 0cm;
background-color: #ffffff;
}
span > #tab:mytable {
  font-weight: bold;
}
.section {
  padding-left: 10mm;
  padding-right: 10mm;
}
.main p {
  padding-left: 30mm;
  padding-right: 30mm;
}
.main_pic {
  width: 100%;
  display: block;
  margin-left: auto;
  margin-right: auto;
}
.orcid img {
  width: 3%;
}
</style>
</head>
<body>


<div class="poster_wrap">
<div class="column outer">
<br>
<div class="title section">
<h1 id="title">cILR: Competitive Isometric Log Ratio for Taxonomic Enrichment Analysis</h1>
<br>
<h3 id="author" class="author">

  </h3>

<h5 id="author_extra" class="author_extra">
Quang Nguyen<sup>1 2</sup>
 Anne G. Hoen<sup>1 2</sup>
 H. Robert Frost<sup>2</sup>
</h5>

<p id="affiliation" class="affiliation">
<sup>1</sup> Department of Epidemiology, Dartmouth College<br> <sup>2</sup> Department of Biomedical Data Science, Dartmouth College
</p>

</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<ul>
<li>Microbiome data is sparse, compositional, and high-dimensional, which makes it difficult to apply standard statistical techniques. Set-based aggregation from the genomics literature is a well-defined approach to alleviate some of these issues.<br />
</li>
<li>Standard practice performs aggregation through abundance summation, but using summation doesnâ€™t allow for comparison of enrichment across sets of different sizes.<br />
</li>
<li>Here we developed a microbiome specific set-based aggregation method generating enrichment scores at the sample level. We address the compositionality of microbiome data by using the isometric log ratio transformation. Our score is equivalent to a competitive null hypothesis by defining enrichment as a comparison between taxa inside the set and its complement.</li>
</ul>
</div>
<div id="acknowledgements" class="section level2">
<h2>Acknowledgements</h2>
<p>This research is supported by funding from the National Institutes of Health (grants NLM R01LM012723, NIGMS P20GM104416, NLM K01LM012426 NIH UG3 OD023275, NIEHS P01ES022832 and EPA RD-83544201).</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="sample-level-significance-testing" class="section level3">
<h3>Sample-level significance testing</h3>
<div class="figure" style="text-align: left"><span id="fig:figure1"></span>
<img src="figures/sim_data_ss_hypo.png" alt="Type I error rate **(A)**, and power **(B)** across different parametric simulation scenarios. Confidence bounds were obtained using Agresti-Couli approach. False-positive rate **(C)** and true positive rate **(D)** evaluation of similar methods on real 16S rRNA data from the oral microbiome of the gingival site. For **(A)** and **(B)**, enrichment of a specified set was tested at the sample level using cILR and the Wilcoxon rank sum test at $\alpha$ of 0.05. For **(C)** and **(D)**, the set of aerobic microbes was tested for enrichment in all samples and was identified as correctly enriched if a significant $p$-value was obtained in supragingival samples." width="100%" />
<p class="caption">
Figure 1: Type I error rate <strong>(A)</strong>, and power <strong>(B)</strong> across different parametric simulation scenarios. Confidence bounds were obtained using Agresti-Couli approach. False-positive rate <strong>(C)</strong> and true positive rate <strong>(D)</strong> evaluation of similar methods on real 16S rRNA data from the oral microbiome of the gingival site. For <strong>(A)</strong> and <strong>(B)</strong>, enrichment of a specified set was tested at the sample level using cILR and the Wilcoxon rank sum test at <span class="math inline">\(\alpha\)</span> of 0.05. For <strong>(C)</strong> and <strong>(D)</strong>, the set of aerobic microbes was tested for enrichment in all samples and was identified as correctly enriched if a significant <span class="math inline">\(p\)</span>-value was obtained in supragingival samples.
</p>
</div>
</div>
<div id="predictive-models" class="section level3">
<h3>Predictive models</h3>
<div class="figure" style="text-align: left"><span id="fig:figure2"></span>
<img src="figures/data_prediction_plot.png" alt="Classification performance of a naive random forest model using cILR scores as inputs compared against existing methods in gene set testing literature and the standard centered-log ratio transformation approach. The learning task involves predicting patients with inflammatory bowel disease (including Crohn's disease and ulcerative colitis) versus controls. Data sets used span both 16S rRNA sequencing (Gevers et al.) and whole genome shotgun sequencing (Nielsen et al.)" width="100%" />
<p class="caption">
Figure 2: Classification performance of a naive random forest model using cILR scores as inputs compared against existing methods in gene set testing literature and the standard centered-log ratio transformation approach. The learning task involves predicting patients with inflammatory bowel disease (including Crohnâ€™s disease and ulcerative colitis) versus controls. Data sets used span both 16S rRNA sequencing (Gevers et al.) and whole genome shotgun sequencing (Nielsen et al.)
</p>
</div>
</div>
<div id="differential-abundance-analysis" class="section level3">
<h3>Differential abundance analysis</h3>
<div class="figure" style="text-align: left"><span id="fig:figure3"></span>
<img src="figures/data_diff_ab.png" alt="Differential abundance analysis using corncob, DESeq2 and cILR with either Wilcoxon rank sum test or Welch's t-test. Panel \textbf{(A)} shows type I error results as the proportion of significant genera after 500 iterations where case/control status was assigned randomly to each sample. Panel \textbf{(B)} shows true positive rate results as the proportion of significant genera who are either obligate anaerobes or aerobes. Both evaluations use 16S rRNA gene sequencing data from HMP. Type I error evaluation used stool samples while the true positive rate evaluation used samples from the gingival site." width="100%" />
<p class="caption">
Figure 3: Differential abundance analysis using corncob, DESeq2 and cILR with either Wilcoxon rank sum test or Welchâ€™s t-test. Panel  shows type I error results as the proportion of significant genera after 500 iterations where case/control status was assigned randomly to each sample. Panel  shows true positive rate results as the proportion of significant genera who are either obligate anaerobes or aerobes. Both evaluations use 16S rRNA gene sequencing data from HMP. Type I error evaluation used stool samples while the true positive rate evaluation used samples from the gingival site.
</p>
</div>
</div>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<p><img src="poster_diagrams_white.png" width="100%" style="display: block; margin: auto;" /></p>
</div>

</div>
<div class="main">
<p>Our method leverages the <strong>isometric log ratio transformation</strong> to <strong>generate enrichment scores for taxa sets</strong> that can be used for standard microbiome analyses while also allowing for <strong>sample-level significance testing</strong> under a <strong>competitive null hypothesis</strong><br><br> <img src="nice_graphic.png" class="main_pic" /></p>


</div>
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
var script = document.createElement("script");
script.type = "text/javascript";
var src = "true";
if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
if (location.protocol !== "file:" && /^https?:/.test(src))
src = src.replace(/^https?:/, '');
script.src = src;
document.getElementsByTagName("head")[0].appendChild(script);
})();
</script>


</body>
</html>
